src/charm.py

```diff
@@ -41,83 +70,51 @@ def __init__(self, *args):
 
     def _update_prometheus_jobs(self, _):
         """Setup Prometheus scrape configuration for external targets."""
-        self.unit.status = ActiveStatus()
         if not self.unit.is_leader():
             return
 
-        jobs = self._scrape_jobs()
-        if not jobs:
-            return
+        if jobs := self._scrape_jobs():
+            for relation in self.model.relations[self._prometheus_relation]:
+                relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
 
-        for relation in self.model.relations[self._prometheus_relation]:
-            relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
+            self.unit.status = ActiveStatus()
 
-    def _scrape_jobs(self):
-        targets = self._targets()
+    def _scrape_jobs(self) -> list:
+        if targets := self._targets():
+            static_config = {"targets": targets}
+            if labels := self._labels():
+                static_config.update(labels=labels)
```

Comment:
Apart from deep nesting I do not see how this address the issue of empty labels I posted in my previous review. 

```
import this
...
Flat is better than nested.
...
```

Reply:
Nested code is harder to understand, once you get past metal blocks against (inverted) negative logic.

Reply:
With this change, no "lables: {}" is created. Tests were updated and confirm this.